PWGHF/TableProducer/candidateSelectorD0.cxx

```diff
@@ -357,33 +363,40 @@ struct HfCandidateSelectorD0 {
 
       if (applyMl) {
         // ML selections
+        bool isSelectedMlD0 = false;
+        bool isSelectedMlD0bar = false;
 
-        std::vector<float> inputFeatures{candidate.cpa(),
-                                         candidate.cpaXY(),
-                                         candidate.decayLength(),
-                                         candidate.decayLengthXY(),
-                                         candidate.impactParameter0(),
-                                         candidate.impactParameter1(),
-                                         candidate.impactParameterProduct()};
-
-        bool isSelectedMl = hfMlResponse.isSelectedMl(inputFeatures, ptCand, outputMl);
-        hfMlD0Candidate(outputMl);
+        if (statusD0 > 0) {
+          std::vector<float> inputFeaturesD0 = hfMlResponse.getInputFeatures(candidate, trackPos, trackNeg, o2::analysis::pdg::kD0);
+          isSelectedMlD0 = hfMlResponse.isSelectedMl(inputFeaturesD0, ptCand, outputMlD0);
+        }
+        if (statusD0bar > 0) {
+          std::vector<float> inputFeaturesD0bar = hfMlResponse.getInputFeatures(candidate, trackPos, trackNeg, o2::analysis::pdg::kD0Bar);
+          isSelectedMlD0bar = hfMlResponse.isSelectedMl(inputFeaturesD0bar, ptCand, outputMlD0bar);
+        }
 
-        if (!isSelectedMl) {
+        if (!isSelectedMlD0) {
           statusD0 = 0;
+        }
+        if (!isSelectedMlD0bar) {
           statusD0bar = 0;
         }
+
+        if (isSelectedMlD0 || isSelectedMlD0bar) {
+          hfMlD0Candidate(outputMlD0, outputMlD0bar);
+        }
```

Comment:
You should always fill the table, otherwise it will be not joinable with those of the candidates and selection flag.